flask/testing.py

```diff
@@ -33,6 +34,18 @@ def make_test_environ_builder(app, path='/', base_url=None, *args, **kwargs):
             path = url.path
             if url.query:
                 path += '?' + url.query
+
+    if json:
```

Comment:
You really want `if json is not None` here. Otherwise `json={}` won't result in a JSON request.